### PR TITLE
  * mod_http2: synchronizes sources with trunk

### PR DESCRIPTION
**Pull request by icing**
Complete synchronisation of sources with trunk version.

  *) mod_http2: when a server is restarted gracefully, any idle h2 worker
     threads are shut down immediately.
     Also, change OpenSSL API use for deprecations in OpenSLL 3.0.

to verify, run

```
> make install
> pytest
```

### CODE REVIEW

File: modules/http2/h2_version.h [hunk 1]
```diff
@@ -27,15 +27,15 @@
  * @macro
  * Version number of the http2 module as c string
  */
-#define MOD_HTTP2_VERSION "1.15.18"
+#define MOD_HTTP2_VERSION "1.15.24-git"
```
Comment (rpluem) [inline modules/http2/h2_version.h]: Is this the correct version number?

File: modules/http2/h2_version.h [hunk 2]
```diff
@@ -27,15 +27,15 @@
...
 
 /**
  * @macro
  * Numerical representation of the version number of the http2 module
  * release. This is a 24 bit number with 8 bits for major number, 8 bits
  * for minor and 8 bits for patch. Version 1.2.3 becomes 0x010203.
  */
-#define MOD_HTTP2_VERSION_NUM 0x010f12
+#define MOD_HTTP2_VERSION_NUM 0x010f18
```
Comment (rpluem) [inline modules/http2/h2_version.h:38]: Is this the correct version number?

File: modules/http2/h2_workers.c [hunk 3]
```diff
@@ -271,14 +314,50 @@ static apr_status_t workers_pool_cleanup(void *data)
...
+                         "still seeing %d workers living",
+                         apr_atomic_read32(&workers->worker_count));
+            continue;
+        }
+    }
+    if (i >= n) {
+        ap_log_error(APLOG_MARK, APLOG_WARNING, 0, workers->s,
+                     APLOGNO(10291) "h2_workers: cleanup, %d idle workers "
+                     "did not exit after %d seconds.",
+                     apr_atomic_read32(&workers->worker_count), i);
```
Comment (rpluem) [inline modules/http2/h2_workers.c]: Isn't this i*timeout seconds (provided timeout is in seconds)?

File: modules/http2/h2_version.h (hunk 1 above)
Comment (icing) [reply modules/http2/h2_version.h]: Fixed by removing suffix. Thanks for noticing.

File: modules/http2/h2_version.h (hunk 2 above)
Comment (icing) [reply modules/http2/h2_version.h:38]: yes, on github updated by configure.

File: modules/http2/h2_workers.c (hunk 3 above)
Comment (icing) [reply modules/http2/h2_workers.c]: Changed the implementation to have an end time and reporting the waited seconds.

**Pull request closed without merging**